mayastor/src/rebuild/rebuild_impl.rs

```diff
@@ -597,11 +599,21 @@ impl RebuildJob {
     }
 
     async fn await_all_tasks(&mut self) {
-        while self.await_one_task().await.is_some() {
-            if self.task_pool.active == 0 {
-                break;
+        debug!(
+            "Awaiting all active tasks({}) for rebuild {}",
+            self.task_pool.active, self.destination
+        );
+        if self.task_pool.active > 0 {
+            while self.await_one_task().await.is_some() {
+                if self.task_pool.active == 0 {
```

Comment:
just a nit. The loop could be more compact:
```
while self.task_pool.active > 0 {
   self.await_one_task().await...
}
```

Reply:
Thanks done, also added an error condition which although unlikely as we never close the channels.